docs/design/autoware-architecture/index.md

```diff
@@ -0,0 +1,43 @@
+# Architecture Overview
+
+This page describes the architecture of Autoware.
+
+## Introduction
+
+The current Autoware is defined to be a layered architecture that clarifies each module's role and simplifies the interface between them. By doing so:
+
+- Autoware's internal processing becomes more transparent
+- Collaborative development is made easier because of the reduced interdependency between modules
+- Users can easily replace an existing module (e.g. localization) with their own software component by simply wrapping their software to fit in with Autoware's interface
+
+Note that the initial focus of this architecture design was solely on driving capability, and so the following features were left as future work:
```

Comment:
I think this is okay since it is an explanation of design decisions that were made in the past, and listing features that were left to be developed later on.